gender_novels/corpus_gen.py

```diff
@@ -111,22 +125,75 @@ def get_publication_date(author, title, id = None):
 
 def get_publication_date_from_copyright(novel_text):
     """
-    Tries to extract the publication date from the copyright statement in the given text
+    Tries to extract the publication date from the copyright statement in the
+    given text
+    Otherwise returns None
+
     >>> novel_text = "This work blah blah blah blah COPYRIGHT, 1894 blah
     >>> novel_text += and they all died."
     >>> from gender_novels import corpus_gen
     >>> get_publication_date_from_copyright(novel_text)
     1894
 
     TODO: should this function take the novel's text as a string or the id or?
-    TODO: should function also try to find publication years not prefaced with "copyright" at
-        the risk of finding arbitrary 4-digit numbers?
+    TODO: should function also try to find publication years not prefaced with
+        "copyright" at the risk of finding arbitrary 4-digit numbers?
     :param novel_text: string
     :return: int
     """
     match = re.search(r"(COPYRIGHT\,*\s*) (\d{4})", novel_text, flags = re.IGNORECASE)
     return match.group(2)
 
+def get_country_publication(author, title):
+    """
+    Tries to get the country of novel
+    @TODO: Country of origin or residence of author?
+
+    >>> from gender_novels import corpus_gen
+    >>> get_country_publication("Hawthorne, Nathaniel", "The Scarlet Letter")
+    'United States'
+
+    :param author: str
+    :param title: str
+    :return: str
+    """
+    # TODO(duan): implement this function
+    pass
+
+def get_author_gender(author):
+    """
+    Tries to get gender of author, 'female', 'male', or 'non-binary'.
+    If it fails returns 'unknown'
+
+    >>> from gender_novels import corpus_gen
+    >>> get_author_gender("Hawthorne, Nathaniel")
+    male
+
+    :param author: str
+    :return: str
+    """
+    # TODO(duan): implement this function
+    pass
+
+def get_subject(author, title, id = None):
+    """
+    Tries to get subjects
+    TODO: Subject as defined by Gutenberg or LoC or what?
+    :param: author: str
+    :param: title: str
+    :param: id: int
+    :return: list
+    """
+    pass
+
+def write_metadata(novel_metadata, path):
+    """
+    Writes a row of metadata for a novel into the csv at path
```

Comment:
can you add a TODO or N.B. that explains that this doesn't yet work.  in general  (approving, but a general principle to follow)